### DISCUSSION TITLE
Why Does Minecraft Crash Without A Log/Crash Report With My Mod?

### DISCUSSION THREAD

**Discussion by mangodev01** [FabricMC/fabric-api#2502] [Mod Dev Support] [answered]
I wanted to add datagen to my mod, idk why is mc crashing, but when i run the game, it crashes immediately with no Log/Crash Report, does anyone know what is going on here?
repository **->** [here](https://github.com/Gamer069/illiatestmod)

**Comment by warjort** [comment]
Look in your run/logs/latest.log and you will see:
````
[15:13:13] [Render thread/ERROR] (Minecraft) Unhandled game exception
java.lang.IllegalStateException: Some intrusive holders were not added to registry: [Reference{null=Block{minecraft:air}}]
        at net.minecraft.util.registry.SimpleRegistry.freeze(SimpleRegistry.java:352) ~[minecraft-project-@-merged-named.jar:?]
````
This is because you create 4 blocks but only register 3 of them.

https://github.com/Gamer069/illiatestmod/blob/ace2dd4e8a1afcf12bb9e3fc3dac58e7f2f922dd/src/main/java/net/illia/illiatestmod/block/ModBlocks.java#L14

**Comment by mangodev01** [comment, marked as answer]
> Look in your run/logs/latest.log and you will see:
> 
> ````
> 
> [15:13:13] [Render thread/ERROR] (Minecraft) Unhandled game exception
> 
> java.lang.IllegalStateException: Some intrusive holders were not added to registry: [Reference{null=Block{minecraft:air}}]
> 
>         at net.minecraft.util.registry.SimpleRegistry.freeze(SimpleRegistry.java:352) ~[minecraft-project-@-merged-named.jar:?]
> 
> ````
> 
> This is because you create 4 blocks but only register 3 of them.
> 
> 
> 
> https://github.com/Gamer069/illiatestmod/blob/ace2dd4e8a1afcf12bb9e3fc3dac58e7f2f922dd/src/main/java/net/illia/illiatestmod/block/ModBlocks.java#L14
> 
> 

Thanks

**Reply by mangodev01** [reply]
it worked, but also how to add lang data gen, there is no provider for it, right?

**Reply by IMB11** [reply]
Not yet - its currently pending a PR #2451

**Reply by IMB11** [reply]
You can copy the related classes if you want to use it now.

**Reply by mangodev01** [reply]
thanks

**Reply by mangodev01** [reply]
but also, how to add block loot table data gen?
